core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java

```diff
@@ -944,39 +975,51 @@ public AbortTransactionResponse call(TrxRegionService instance) throws IOExcepti
-		     String errMsg = new String("Abort of transaction: " + transactionId
+		     String errMsg = new String("Abort of transaction " + transactionId
                           + " participantNum: " + participantNum + " region: " + Bytes.toString(regionName)
                           + " threw Exception: " + exceptionString);
                      LOG.error(errMsg);
                      if(exceptionString.contains("UnknownTransactionException")) {
+                       LOG.error("doAbortx caught and rethrowing UnknownTransactionException for transaction " + transactionId
+                               + " participantNum: " + participantNum + " region: " + Bytes.toString(regionName)); 
                        throw new UnknownTransactionException(errMsg);
                      }
+                     LOG.error("doAbortx throwing RetryTransactionException for transaction " + transactionId
+                               + " participantNum: " + participantNum + " region: " + Bytes.toString(regionName)); 
                      throw new RetryTransactionException(cresponse.getException());
                    }
                  }
                  retry = false;
               }
            }
           catch (UnknownTransactionException ute) {
-             LOG.error("Got unknown exception in doAbortX by participant " + participantNum
-                       + " for transaction: " + transactionId, ute);
-             transactionState.requestPendingCountDec(true);
-             throw ute;
+             String errMsg = new String("Got unknown exception in doAbortX by participant " + participantNum
+                       + " for transaction " + transactionId);
+             if (ignoreUnknownTransaction) {
+                LOG.info(errMsg + " ,but ignored", ute);
+                transactionState.requestPendingCountDec(null);
+             }
+             else {
+                LOG.error(errMsg, ute);
+                transactionState.logUteDetails();
+                transactionState.requestPendingCountDec(null);
+             }
           }
           catch (RetryTransactionException rte) {
               if (retryCount == RETRY_ATTEMPTS) {
                   String errMsg = "Exceeded retry attempts in doAbortX: " + retryCount;
                   LOG.error(errMsg, rte); 
-                  transactionState.requestPendingCountDec(true);
-                  throw new DoNotRetryIOException(errMsg, rte);
+                  DoNotRetryIOException dnre = new DoNotRetryIOException(errMsg, rte);
+                  transactionState.requestPendingCountDec(dnre);
+                  throw dnre;
               }
               else if (rte.toString().contains("Asked to commit a non-pending transaction ")) {
-                 LOG.error(" doCommitX will not retry transaction: " + transactionId , rte);
+                 LOG.error(" doCommitX will not retry transaction " + transactionId , rte);
```

Comment:
May need a requestPendingCountDec here in this case.
Also the Log error above should be doAbortX